### DISCUSSION TITLE
Syncthing not showing folder info

### DISCUSSION THREAD

**Discussion by macbean101** [linuxserver/Heimdall#989] [App issues]
I set up the syncthing app, put in the API key and tested it successfully.  However, the app shows 0 info no matter how much is queued to sync.  I have 2 folders, not sure if this is part of the issue.  

![image](https://user-images.githubusercontent.com/25144645/186614275-73343f7a-8ac1-4e93-ba15-f390f8328bc6.png)

I'm using Docker and its up to date.

**Comment by rousseauusername** [comment]
Hi,

I see in your screenshot that you were able to get Heimdall to connect to the Syncthing API. I am having trouble with this step, getting an error in Heimdall "General error connecting with API".

Would you be able to help with the settings you used to get Heimdall to connect to the Syncthing API? Which URL did you put in Heimdall for Syncthing? Anything after the port number? And is your Syncthing GUI set to use https?

Thank you very much for your help!

**Comment by DasLemo** [comment]
I don't know how to solve the problem with the API connection. For me the connection is working. But I was able to get the initial problem solved. In the Syncthing.php file you have to change the corresponding code to: 

    public function livestats()
    {
        $data = [];
        $needed_files = 0;
        $needed_bytes = 0;
        $status = 'inactive';
        $attrs = $this->get_request_attrs();

        # first get a list of folders
        $res = parent::execute($this->url('/rest/stats/folder'), $attrs);
        $details = json_decode($res->getBody());

        foreach ($details as $folder => $folder_status) {
            $folder_db_res = parent::execute($this->url("/rest/db/status?folder={$folder}"), $attrs);
            $folder_db = json_decode($folder_db_res->getBody());
            
            if ($folder_db) {
                $needed_files += $folder_db->needFiles;
                $needed_bytes += $folder_db->needBytes;
            }     
        }

        if ($needed_files || $needed_bytes) {
            $status = 'active';
        }

        $data['needed_files'] = $needed_files;
        $data['needed_bytes'] = $needed_bytes;
        return parent::getLiveStats($status, $data);
    }


Additionally i wanted my WebUi to show remaining GB instead of bytes. This was done by changing the livestats.blade.php to:

```
<ul class="livestats">
    <li>
        <span class="title">Needed Files</span>
        <strong>{!! $needed_files !!}</strong>
    </li>
    <li>
        <span class="title">Needed Gigabytes</span>
        <strong>{!! round($needed_bytes / 1073741824, 2) !!} GB</strong>
    </li>
</ul>
```

Hope this will be usefull for others as well as the zeros on the WebUI were driving me crazy haha.

**Reply by rousseauusername** [reply]
Thank you very much for your reply.

Any chance you could explain how you set up your Heimdall to access the API? The URL you used for the application (for example I'm trying with https://192.168.1.11:8384/ , nothing after the port number, do you have something?), and did you set a different URL in the optional config section at the bottom? Or did you just put the Syncthing API key there?

Thank you very much!

**Reply by DasLemo** [reply]
In my settings I'm using http. In my Syncthing-Gui-Settings I haven't enabled https. Have you tried using http://192.168.1.11:8384/ instead of https? I just inserted the API-Key from Syncthing I generated under settings-->general into Heimdall and when I press the test-button I get a success message from Heimdall. Alternatively you can use curl -H "X-API-Key: YOUR_API_KEY" 192.168.1.11:8384/rest/stats/folder to see if your Syncthing API is working correctly.
<img width="754" alt="Screenshot 2023-04-07 183124" src="https://user-images.githubusercontent.com/51260230/230645583-b180dfc8-9498-4fc7-8660-51aa1e1ddb4e.png">
.

**Reply by rousseauusername** [reply]
Thank you for the feedback!

I tried with just http instead of https, but it still doesn't work, Heimdall gives me an error "General error connecting with API".

I also tried in Terminal with the command you provided and get a message saying "Temporary Redirect" that I don't know how to interpret if it means the API is working or not?

I have attached screenshots in both cases.

Thank you again for all your help!

<img width="1277" alt="Screenshot 2023-04-07 at 16 44 52" src="https://user-images.githubusercontent.com/127876647/230676732-8b7a0afc-8147-4802-b002-a245d1855c89.png">

<img width="693" alt="Screenshot 2023-04-07 at 16 47 35" src="https://user-images.githubusercontent.com/127876647/230676811-005fe112-1965-46e4-b7ae-66f876b7b451.png">

**Reply by macbean101** [reply]
> Hope this will be usefull for others as well as the zeros on the WebUI were driving me crazy haha.

Thanks!  worked like a charm.

**Comment by DasLemo** [comment]
Oh I'm sorry. I cannot help you with this issue tbh. It seems like this is a problem with your Syncthing installation when the API request isn't returning the expected information. The returned information should look something like this:
```
{
  "xyz": {
    "lastFile": {
      "at": "2023-03-03T01:10:28+01:00",
      "filename": "xyz",
      "deleted": false
    },
    "lastScan": "2023-04-08T18:16:45+02:00"
  },
  "xyz": {
    "lastFile": {
      "at": "2023-04-08T09:35:06+02:00",
      "filename": "xyz",
      "deleted": false
    },
    "lastScan": "2023-04-08T18:56:41+02:00"
  },
  "xyz": {
    "lastFile": {
      "at": "2023-04-08T05:01:59+02:00",
      "filename": "xyz",
      "deleted": true
    },
    "lastScan": "2023-04-08T18:27:25+02:00"
  }
}
```
Heimdall seems to be working correctly. What is your installation setup? I have everything installed in Docker on Unraid. I don't think I changed any of the default syncthing settings if I remeber correctly. Maybe you should start from the beginning with your Syncthing installtion and focus on getting the API to work. But if you would like to keep trying with your current installtion it would be better if you asked in the Syncthing forum for help as I would think they would know what's wrong with your installtion. 
I'm sorry i cannot be a better help for you. But I don't know enough about Syncthing and it's API as I am using it for a very basic task... Good luck to you anyway.

**Reply by rousseauusername** [reply]
Ah, that's great info, thank you very much!

I will work on getting Syncthing's API properly setup.

Thanks again for your help!

**Comment by ShinyYellowBanana** [comment]
In Syncthing disable TLS in Actions>Advanced  and use http:// instead of https://

**Reply by macbean101** [reply]
This wasn't the issue for me.  since the webui is only used on my home network, i was already using http.

But the revised code snippet provided above fixed my issue.
